File: android/domain/src/main/java/com/ex/lolcompose/domain/model/Champion.kt
package com.ex.lolcompose.domain.model

data class Champion(
    val id : String = "",
    val name : String = "",
    val title : String = "",
    val lore : String = "",
    val tags : List<String>? = null,
    val skins : List<Skin>? = null,
    val spells : List<Spell>? = null,
    val passive : Passive? = null
)

File: flutter/lib/features/champion/domain/entity/champion.dart
import 'package:lol_champion/features/champion/domain/entity/passive.dart';
import 'package:lol_champion/features/champion/domain/entity/skin.dart';
import 'package:lol_champion/features/champion/domain/entity/spell.dart';

class Champion {
  const Champion({
    required this.id,
    required this.name,
    required this.title,
    required this.lore,
    required this.tags,
    required this.skins,
    required this.spells,
    required this.passive,
  });

  final String id;
  final String name;
  final String title;
  final String lore;
  final List<String> tags;
  final List<Skin> skins;
  final List<Spell> spells;
  final Passive? passive;
}
